consumers/on-demand.md

```diff
@@ -0,0 +1,44 @@
+In contrast, Pyth Network does not operate such a process and instead delegates this work to Pyth Network users.
+Pyth price updates are streamed off-chain via the Wormhole Network, a cross-chain messaging protocol.
+These updates are signed in such a way that the Pyth on-chain program can verify their authenticity.
+Typically, users of Pyth Network prices will submit a single transaction that simultaneously updates the price and uses it.
+
+Pyth Network's on-demand price update model has many advantages over the (more common) push model:
+
+- **Gas efficiency** -- On-chain prices are only updated when they are needed.
+  In the push model, the oracle can waste gas by submitting price updates that no one will use.
+  Furthermore, the cost of updating the oracle is distributed amongst its users instead of borne entirely by Pyth Network.
+  The cost of maintaining the on-chain prices can be substantial for a single entity, but is minimal when spread across all users.
+  Many of the subsequent advantages follow from the fact that Pyth Network does not have to pay gas fees for every single update.
+- **High update frequency** -- Pyth Network price feeds update once per second, which is faster than the blocktime of most blockchains.
+  Such frequent updates would not be possible if every price had to be pushed on-chain.
+  However, push oracles typically update even less frequently than the blocktime, because it is simply too expensive to update feeds more frequently.
+- **Low latency** -- Every transaction can use a recent off-chain price, instead of relying on the last on-chain update pushed by the oracle itself.
+- **More price feeds** -- Pyth Network can scale to thousands of price feeds due to its gas efficiency.
+  The oracle incurs no added costs for each additional feed, and users pay gas costs for new feeds only when those feeds are used on-chain.
+- **Common infrastructure** -- Every component of Pyth Network is *shared across blockchains* except for the contract deployed on the destination chain.
+  These shared components can therefore be built with high reliability and accuracy targets, benefitting every chain the oracle is deployed on.
+  This approach also allows Pyth Network to rapidly launch on new blockchains and ecosystems with all of the existing price feeds.
+- **Sustainable** -- The Pyth Network protocol has been designed to allow for the optional enablement of data fees to update the state of an on-chain price feed.
+  These fees will compensate data providers for their effort and motivates them to contribute additional data.
+  Oracles without such a mechanism are inherently unsustainable and likely to fail if the operating organization runs out of money.
+
+These are substantial advantages that allow Pyth Network feeds to operate faster and more scalably than other oracles'.
+However, there are also some disadvantages of the on-demand model:
```

Comment:
The bullet-points in the disadvantages section are arguably more important for integrators to be aware of than those in the advantages section. I think we should split the disadvantages section up into several sub-sections, and structure the page like this:

- Overview (with a diagram)
- Advantages
- Integration (talk about how easy the SDKs are to use)
- Adversarial Selection
- End Users Transaction Cost
- Update Fees

This avoids the use of the word disadvantages, and highlights the important information. Each section should have a header so it looks nice in the sidebar menu.